feat: add homepage admin test

naxoc opened the pull request:
This test is very simple. Maybe even too simple :-D 

We could use it on a sites without Woo installed to see if we can trigger errors that would be caused by code assuming Woo.

naxoc commented:
It looks like the failing test is not from the one just added.

adekbadek reviewed (commented):
Nice! That's a good exercise in e2e :)

> We could use it on a sites without Woo installed to see if we can trigger errors that would be caused by code assuming Woo.

Were you able to trigger any such error? What this PR adds is insubstantial without this context. I'd rather we add it along with the with/without Woo infrastructure, and with knowledge of what kind of errors are we trying to catch here.

naxoc commented:
I documented the dependencies between tests a bit and added a diagram to README.

The CI now does not need to reset the site – switching between snapshots is built into the test. I've kept the reset file in the repo for local testing work. 

I have trouble getting the `donations.spec.ts` test to succeed locally. Not sure why there is a difference, but I can fix it in another PR if necessary.

I have not set up the snapshots on e2e.newspackstaging.com yet – I'd like to know that we are going in the right direction with this first.

adekbadek commented:
> I have trouble getting the donations.spec.ts test to succeed locally. 

I'm running into issues with the snapshots themselves – after loading a snapshot, `wp_user_roles` option gets borked. I have to look into it. 

---

In the meantime, the dependencies look wrong:

<img width="387" alt="image" src="https://github.com/user-attachments/assets/b2570151-349d-4868-861b-66361a0969e1" />

All tests should run on **both** vanilla and with-woo setups, apart from the donations test. All tests besides donations should have both tags:

<img width="602" alt="image" src="https://github.com/user-attachments/assets/7cf43b86-e1be-4cd4-91ff-f1ac17b40ef3" />

naxoc commented:
> All tests should run on both vanilla and with-woo setups

Ah, didn't know that. I've tagged all but the donations one. 

> I'm running into issues with the snapshots themselves – after loading a snapshot, wp_user_roles option gets borked.

Does this happen only when running the test or also if you do it with the UI or CLI?

naxoc commented:
The `reader-registration.spec.ts` test needs woo-memberships (for the login button), so I removed the `@vanilla` tag from that again.

I also added a little orange label thingy at the left bottom of all pages. That should help in debugging tests to see what snapshot is in use. 
![snapshot-label](https://github.com/user-attachments/assets/f4ba59a5-8bb2-4c48-902c-7844341321ca)

adekbadek commented on README.md:
```diff
@@ -28,22 +28,72 @@ Will need a publicly accessible (or at least accessible for the CI server) test
 If the tests manipulate any persistent items (anything in the DB), reset commands should be added to the `/bin/e2e-reset.sh` script. In the future, if that's too brittle, we might opt for a full reset, though.
+
+## Resetting the test site
+The test site can be reset by running the `e2e-reset.sh` script. Please note that on initial setup of the site, you will need to set up the Stripe test connection manually or put this in
```
> put this in

Put what in?

naxoc replied:
Woops. Thanks!

adekbadek commented:
> > I'm running into issues with the snapshots themselves – after loading a snapshot, wp_user_roles option gets borked.

> Does this happen only when running the test or also if you do it with the UI or CLI?

Both CLI and UI

naxoc commented:
I added the snapshots (with the correct names this time). If we are happy with the snapshot approach we can delete the reset step on CI.

The test is still failing though. Not sure I understand why it's not finding the snapshots. They are there and I can't reproduce this locally. @adekbadek does it have to do with what is deployed to the e2e test server? I'm running out of ideas here.

adekbadek commented:
If you look at the artifacts on [CI run](https://app.circleci.com/pipelines/github/Automattic/newspack-e2e-tests/1462/workflows/2e63748b-fe3d-418f-920a-2f572a87245c/jobs/1687/steps)

<img width="501" alt="image" src="https://github.com/user-attachments/assets/5c6f724f-012f-4242-b33c-c7ffe6a9d704" />

you'll find videos. The test if failing because it can't access the `/wp-admin/tools.php?page=newspack-snapshots` page:

<img width="538" alt="image" src="https://github.com/user-attachments/assets/82570866-4a52-4837-bd87-647c3861dff5" />

Which is because of [this check](https://github.com/Automattic/newspack-manager/blob/37bf560f7a859490cad30afb92ac61a2a8f8815d/includes/class-utils.php#L47) in `newspack-manager`.

`newspack-manager` needs to be updated to allow bypassing the `can_do_something_very_serious` with an env variable. Or something else.

naxoc commented:
Thanks for the hint @adekbadek ! This got in yesterday: https://github.com/Automattic/newspack-manager/pull/373 and the test passes now.

adekbadek reviewed (approved):
🙌
